### DISCUSSION TITLE
Average data by day of week

### DISCUSSION THREAD

**Discussion by metmarkosaric** [plausible/analytics#754] [Anything Else]
"Are there any views of the data by day of week (e.g. average over a longer period)"

(feedback via email)

**Comment by ukutaht** [comment]
These sorts of heatmaps seem like the most intuitive way to visualize it: http://www.dartistics.com/adobeanalytics/int-heatmap.html

**Reply by metmarkosaric** [reply]
Yeah those look nice as they also do hour of the day which is another cool metric for some use cases. If we can add average number of visitors on a specific day of the week to the heatmaps would be even better.

**Reply by Vigasaurus** [reply]
Yeah I think the way GA does it is quite nice; it would definitely be nicer if as Marko said, we can also include the daily total averages as an additional row or similar. I do think a big thing to keep in mind or consider once more reports start getting added (this, web vitals) - is a way for a user to re-order, or hide specific reports. 

![Screenshot_20210119-194631.png](https://user-images.githubusercontent.com/19434157/109164647-49850a00-7740-11eb-8382-24726fdc784c.png)

**Reply by metmarkosaric** [reply]
Yeah looks nice! Some of these ideas such as day of week and web vitals are definitely starting to feel that they should not be displayed by default but more as either a new section in the top menu, or some kind of alerting system that only tells you when you need to know or it only displays when you do some specific selection or filter on the default dashboard. We'll have to think of the design carefully before we introduce them

**Reply by Vigasaurus** [reply]
I think adding a top menu is a dangerous precedent personally, it could easily become like GA/Matomo's left menu over time, and I think the whole one-page model is definitely a big selling point for Plausible. 

My thought was something where the default view will stay like it is now, but a user would be able to enter a so called dashboard "edit mode", where they can drag&drop and rearrange the reports, and select/deselect modules to show on a site by site basis. This could obviously have an alert similar to enabling weekly reports to tell a user that this option is available/`x modules are not shown`. Thoughts?

**Reply by metmarkosaric** [reply]
Yeah Uku was talking about that. That was his intention with the dashboard from when he started building Plausible. Could be cool but sounds like a very big project?

**Reply by Vigasaurus** [reply]
I don't think it'd ultimately be that big, there have been a lot of new and fantastic entries into the React drag&drop library space, and the whole toggling bit is not that big (it'd just need decisions on how/where that is synced to, i.e. only localStorage, to an account, etc). 

It'd not be trivial of course, but I'd say it's less work than stuff like actually adding the web vitals report, doing cities, or most any of the large-scale tasks. If I had to equate it to something, I'd say it's closest to Dark Mode in that it'd be a largely front-end only set of work, with a tiny amount of backend for saving etc.

**Reply by metmarkosaric** [reply]
sounds pretty cool and sorts some of the design issues we might have by adding extra features! definitely something that Uku will talk to you more about

**Reply by captainrobbo** [reply]
That's very nice, but it may be simpler for you to just show a bar or line chart going by hours (7x24=168 data points) and this would certainly meet my use case.  This would still give very simple and intuitive clues as to when a site's users are most active.   Most human activities follow a weekly cycle!

**Reply by ukutaht** [reply]
@Vigasaurus @metmarkosaric 

Yeah since the very beginning I've wanted a UI sort of like the iPhone home screen where you can enter into a drag and drop mode and compose your own dashboard with the reports that you're interested in. We can offer a nice selection of options but in the end the user knows best for themselves whether they're interested in 'day of week' report, for example. This configuration should definitely be synced to the account, not the browser.

I agree that adding a menu could lead to a complex product in the end. It would be a big decision to add a menu and it could introduce a kitchen sink of additional pointless reports. But it could also help compartmentalize things, specifically to separate live traffic stats from background monitors.

The reason it's difficult for me to mix traffic stats from our JS tracker with stats from background monitors is that they are not governed by the same filtering rules as the other reports. We wouldn't be able to show web vitals for `browser=Chrome&source=Google` for example.

I suppose we could start by adding the additional reports to the main dashboard and when filters are used we could add a notice 'filters do not apply to monitoring data' or something like.

In the future if we decide to add more background monitors (accessibility reports, uptime, SEO stuff) then it might feel very natural to break them off from the traffic stats captured from JS.

**Reply by Vigasaurus** [reply]
Hmm yeah I didn't think about the third-party stuff which wouldn't have nice and clean access to filters; I do think that splitting those off is likely better than trying to intermingle filtered/unfiltered data on the same screen. From the current set of ideas it seems like all of the additional stuff would be entirely non-stat related, so maybe just a different view of "Site Suggestions" or similar would be best. (Also I'd imagine that enabling all those background monitors would be opt-in anyways, so they don't need to be super surfaced - nor discoverable outside of suggesting opt-in -  for normal users, anyways)

**Reply by ukutaht** [reply]
> Also I'd imagine that enabling all those background monitors would be opt-in anyways, so they don't need to be super surfaced - nor discoverable outside of suggesting opt-in - for normal users, anyways

Exactly. We'll probably make it a paid extra to mitigate hardware costs. It will be very much an opt-in thing that we don't expect most people to use.

**Comment by luxaritas** [comment]
Being able to see the amount of traffic we're getting by hour/day of week would be super useful for us when doing things like planning when to do maintenance that requires downtime to be least disruptive
